internal/service/alarms/api/openapi.yaml

```diff
@@ -0,0 +1,756 @@
+    description: Probable cause information
+  - name: internal
+    description: |
+      Internal endpoints. This is tag is also useful generating client code.
+
+paths:
+  /O2ims_infrastructureMonitoring/v1/alarms:
+    get:
+      operationId: GetAlarms
+      summary: Retrieve the list of alarms
+      tags:
+        - alarms
+      parameters:
+        - in: query
+          name: filter
+          schema:
+            type: string
+          allowReserved: true
+          description: Optional filter parameter. Only the attributes in AlarmEventRecord are filterable.
+          example: (eq,perceivedSeverity,0)
+      responses:
+        '200':
+          description: Successful response
+          content:
+            application/json:
+              schema:
+                type: array
+                items:
+                  $ref: '#/components/schemas/AlarmEventRecord'
+        '400':
+          description: Bad request
+          content:
+            application/problem+json:
+              schema:
+                $ref: '#/components/schemas/ProblemDetails'
+        '403':
+          description: Forbidden.
+          content:
+            application/problem+json:
+              schema:
+                $ref: '#/components/schemas/ProblemDetails'
+        '404':
+          description: Content not found.
+          content:
+            application/problem+json:
+              schema:
+                $ref: '#/components/schemas/ProblemDetails'
+        '500':
+          description: Internal server error.
+          content:
+            application/problem+json:
+              schema:
+                $ref: '#/components/schemas/ProblemDetails'
+
+  /O2ims_infrastructureMonitoring/v1/alarms/{alarmEventRecordId}:
+    get:
+      operationId: GetAlarm
+      summary: Retrieve exactly one alarm
+      tags:
+        - alarms
+      parameters:
+        - in: path
+          name: alarmEventRecordId
+          required: true
+          schema:
+            type: string
+            format: uuid
+      responses:
+        '200':
+          description: Successful response
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/AlarmEventRecord'
+        '404':
+          description: The specified AlarmEventRecord was not found.
+          content:
+            application/problem+json:
+              schema:
+                $ref: '#/components/schemas/ProblemDetails'
+
+    patch:
+      operationId: AckAlarm
+      summary: Modify exactly one alarm to acknowledge
+      tags:
+        - alarms
+      parameters:
+        - in: path
+          name: alarmEventRecordId
+          required: true
+          schema:
+            type: string
+            format: uuid
+      requestBody:
+        required: true
+        content:
+          application/json:
+            schema:
+              $ref: '#/components/schemas/AlarmEventRecordModifications'
+      responses:
```

Comment:
Specification mentions 409 and 412 codes. Also 400 is important here

Reply:
added